pysma/__init__.py

```diff
@@ -252,39 +294,11 @@ def close_session(self):
     def read(self, sensors):
         """Read a set of keys."""
         payload = {"destDev": [], "keys": list(set([s.key for s in sensors]))}
-        if self.sma_sid is None:
-            yield from self.new_session()
-            if self.sma_sid is None:
-                return False
-        body = yield from self._fetch_json(URL_VALUES, payload=payload)
-
-        # On the first error we close the session which will re-login
-        err = body.get("err")
-        if err is not None:
-            _LOGGER.warning(
-                "%s: error detected, closing session to force another login attempt, got: %s",
-                self._url,
-                body,
-            )
-            yield from self.close_session()
+        result_body = yield from self._read_body(sensors, URL_VALUES, payload)
+        if not result_body:
+            _LOGGER.warning("No values for logging sensors %s", str(list(set([s.key for s in sensors]))))
```

Comment:
~~logging~~ & duplicates here won't hurt (ideally it should be de-duplicated by the library caller)

"No values for sensors: %s", ",".join([s.key for s in sensors])